pkg/chains/formats/slsa/v2alpha2/internal/resolved_dependencies/resolved_dependencies.go

```diff
@@ -0,0 +1,227 @@
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package resolveddependencies
+
+import (
+	"context"
+	"encoding/json"
+
+	"github.com/in-toto/in-toto-golang/in_toto/slsa_provenance/common"
+	v1 "github.com/in-toto/in-toto-golang/in_toto/slsa_provenance/v1"
+	"github.com/tektoncd/chains/pkg/artifacts"
+	"github.com/tektoncd/chains/pkg/chains/formats/slsa/internal/material"
+	"github.com/tektoncd/chains/pkg/chains/objects"
+	"go.uber.org/zap"
+	"knative.dev/pkg/logging"
+)
+
+const (
+	// pipelineConfigName is the name of the resolved dependency of the pipelineRef.
+	pipelineConfigName = "pipeline"
+	// taskConfigName is the name of the resolved dependency of the top level taskRef.
+	taskConfigName = "task"
+	// pipelineTaskConfigName is the name of the resolved dependency of the pipeline task.
+	pipelineTaskConfigName = "pipelineTask"
+	// inputResultName is the name of the resolved dependency generated from Type hinted ARTIFACT_INPUTS results
+	inputResultName = "inputs/result"
+	// pipelineResourceName is the name of the resolved dependency of pipeline resource.
+	pipelineResourceName = "pipelineResource"
+)
+
+// TaskRun constructs `predicate.resolvedDependencies` section by collecting all the artifacts that influence a taskrun such as source code repo and step&sidecar base images.
+func TaskRun(ctx context.Context, tro *objects.TaskRunObject) ([]*v1.ResourceDescriptor, error) {
+	var resolvedDependencies []*v1.ResourceDescriptor
+	var err error
+
+	// add top level task config
+	if p := tro.Status.Provenance; p != nil && p.RefSource != nil {
+		rd := v1.ResourceDescriptor{
+			Name:   taskConfigName,
+			URI:    p.RefSource.URI,
+			Digest: p.RefSource.Digest,
+		}
+		resolvedDependencies = append(resolvedDependencies, &rd)
+	}
+
+	mats := []common.ProvenanceMaterial{}
+
+	// add step images
+	if err := material.AddStepImagesToMaterials(tro.Status.Steps, &mats); err != nil {
+		return nil, err
+	}
+
+	// add sidecar images
+	if err := material.AddSidecarImagesToMaterials(tro.Status.Sidecars, &mats); err != nil {
+		return nil, err
+	}
+
+	// convert materials to resolved dependencies
+	resolvedDependencies = append(resolvedDependencies, convertMaterialsToResolvedDependencies(mats, "")...)
+
+	// add input artifacts
+	mats = artifacts.RetrieveMaterialsFromStructuredResults(ctx, tro, artifacts.ArtifactsInputsResultName)
+	// convert materials to resolved dependencies
+	resolvedDependencies = append(resolvedDependencies, convertMaterialsToResolvedDependencies(mats, inputResultName)...)
+
+	// add task resources
+	mats = material.AddTaskResourcesToMaterials(ctx, tro, []common.ProvenanceMaterial{})
+	// convert materials to resolved dependencies
+	resolvedDependencies = append(resolvedDependencies, convertMaterialsToResolvedDependencies(mats, pipelineResourceName)...)
+
+	// Store git rev as ResolvedDependencies
+	gitCommit, gitURL := material.GitInfo(tro)
```

Comment:
Maybe out of scope of this PR. I think it would be great if we can have a consistent approach across those materials collection functions so that we can first collect all the materials by calling the functions and then call convertMaterialsToResolvedDependencies only once at the end. By saying consistent approach, I mean we either mutate the materials passed through params or return a new list of materials from each function.

Reply:
That's hard since we name the resolved dependencies. So we don't know which material is which once we combine them all.

Reply:
The issue is that materials does not have a name field which breaks the consistency.